File: typscript/2025/February/28/28.ts
function findMedianSortedArrays(nums1: number[], nums2: number[]): number {
    const combined = [...nums1, ...nums2];
    combined.sort((a, b) => a - b);
    const len = combined.length;

    if (len % 2 === 0) {
        const total = combined[Math.floor(len / 2) - 1] + combined[Math.floor(len / 2)];
        return total / 2;
    } else {
        return combined[Math.floor(len / 2)];
    }
}


console.log(findMedianSortedArrays([1, 3], [2]));

File: python/2025/February/28/28.py
def findMedianSortedArrays(nums1, nums2):
    combined = nums1 + nums2
    combined.sort()
    length = len(combined)

    if length % 2 == 0:
        total = combined[length // 2 - 1] + combined[length // 2]
        return total / 2
    else:
        return combined[length // 2]

findMedianSortedArrays([1, 3], [2])
findMedianSortedArrays([1, 2], [3, 4])
